pkg/reconciler/knativeserving/ingress/ingress_service.go

```diff
@@ -30,23 +34,46 @@ import (
 // invalid in Kubernetes 1.20+.
 func IngressServiceTransform(ks *v1beta1.KnativeServing) mf.Transformer {
 	return func(u *unstructured.Unstructured) error {
-		if u.GetAPIVersion() == "v1" && u.GetKind() == "Service" && u.GetName() == "knative-local-gateway" {
-			u.SetNamespace("istio-system")
-			u.SetOwnerReferences(nil)
-			config := ks.GetSpec().GetConfig()
-			if data, ok := config["istio"]; ok {
-				UpdateNamespace(u, data, ks.GetNamespace())
-			}
+		if u.GetAPIVersion() == "v1" && u.GetKind() == "Service" {
```

Comment:
Why not put this into a separate helper function?

Reply:
Same would be for the next two big if statements as they are checks:

...
checkA(...)
checkB(...)
checkC(...)
...